src/Uno.UWP/Globalization/ApplicationLanguages.cs

```diff
@@ -18,11 +19,41 @@ static ApplicationLanguages()
 
 		public static string PrimaryLanguageOverride
 		{
-			get => _primaryLanguageOverride;
+			get
+			{
+				if (_primaryLanguageOverride.Length == 0 &&
+					ApplicationData.Current.LocalSettings.Values.TryGetValue(PrimaryLanguageOverrideSettingKey, out var savedValue) &&
+					savedValue is string stringSavedValue)
+				{
+					_primaryLanguageOverride = stringSavedValue;
+				}
+
+				return _primaryLanguageOverride;
+			}
 			set
 			{
-				_primaryLanguageOverride = value;
+				if (value is null)
+				{
+					throw new ArgumentNullException(nameof(value), "Value cannot be null.");
+				}
+
+				try
+				{
+					// This doesn't match UWP for edge cases. In UWP:
+					// Windows.Globalization.ApplicationLanguages.PrimaryLanguageOverride = "fr-abcdefgh"  // This succeeds.
+					// Windows.Globalization.ApplicationLanguages.PrimaryLanguageOverride = "fr-abcdefghe" // This fails.
+					// However, new CultureInfo(..) fails in both cases.
+					// These are edge cases anyway.
+					_ = new CultureInfo(value);
```

Comment:
I don't know if you usually care to this kind of edge cases which are unlikely to be in a production application. I made my best effort to mostly align with UWP. But not 100% aligning.

Reply:
It's interesting to add the validation anyways. We should be validating this in a runtime or unit test.

Reply:
@jeromelaban This will likely involve restarting the app. I don't know how this is done. Is there an API for that somewhere?

Reply:
Yes, there is but it's not available for all platforms. Let's keep it this way for now.